dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/DataManager.kt

```diff
@@ -0,0 +1,156 @@
+package org.dataland.datalandbackend.service
+
+import org.dataland.datalandbackend.edcClient.api.DefaultApi
+import org.dataland.datalandbackend.interfaces.DataManagerInterface
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataManagerInputToGetData
+import org.dataland.datalandbackend.model.DataMetaInformation
+import org.dataland.datalandbackend.model.StorableDataSet
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+/**
+ * Implementation of a data manager for Dataland including meta data storages
+ */
+@Component("DefaultManager")
+class DataManager(
+    @Autowired var edcClient: DefaultApi
+) : DataManagerInterface {
+    var dataMetaData = mutableMapOf<String, DataMetaInformation>()
+    var companyData = mutableMapOf<String, CompanyMetaInformation>()
+    private var companyCounter = 0
+
+    /*
+    ________________________________
+    Helper-methods:
+    ________________________________
+     */
+
+    private fun verifyCompanyIdExists(companyId: String) {
+        if (!companyData.containsKey(companyId)) {
+            throw IllegalArgumentException("Dataland does not know the company ID $companyId.")
+        }
+    }
+
+    private fun verifyDataIdIsExists(dataId: String) {
+        if (!dataMetaData.containsKey(dataId)) {
+            throw IllegalArgumentException("Dataland does not know the data ID: $dataId.")
+        }
+    }
+
+    private fun verifyDataTypeExists(dataType: String) {
+        val matchesForDataType = dataMetaData.filter { it.value.dataType.equals(dataType) }
+        if (matchesForDataType.isEmpty()) {
+            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
+        }
+    }
```

Comment:
besser:
```
        val matchesForDataType = dataMetaData.any { it.value.dataType.equals(dataType) }
        if (matchesForDataType) {
            throw IllegalArgumentException("Dataland does not know the data type: $dataType")
        }
```
So hört die Suche nämlich auf, wenn das erste passende Element gefunden wurde.

Reply:
👍  Habe ich so gemacht.

Aber in der if-Bedingung habe ich `!matchesForDataType` geschrieben.   Hattest du bestimmt auch gemeint, oder?

Reply:
ja

Reply:
Noch anders: Wie besprochen bitte prüfen, ob Dataland den Datentypen _prinzipiell_ kennt, nicht ob er wirklich in irgendwelchen Daten vorhanden ist.

Reply:
ja - das finde ich sehr gut! Danke.
Solved